src/oci/conmon.rs

```diff
@@ -0,0 +1,292 @@
+//! Interface to [conmon][0], the OCI container runtime monitor.
+//!
+//! [0]: https://github.com/containers/conmon
+
+#![allow(dead_code)]
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use derive_builder::Builder;
+use dyn_clone::{clone_trait_object, DynClone};
+use getset::{Getters, Setters};
+use log::LevelFilter;
+use std::{
+    fmt::{self, Debug},
+    path::{Path, PathBuf},
+    process::Output,
+    string::ToString,
+    time::Duration,
+};
+use strum::AsRefStr;
+use tokio::process::Command;
+
+#[derive(Builder, Debug, Getters, Setters)]
+#[builder(pattern = "owned", setter(into))]
+// Conmon is the main structure to be used when interacting with the container monitor.
+pub struct Conmon {
+    #[getset(get, set)]
+    #[builder(private, default = "Box::new(DefaultExecCommand)")]
+    /// The executor for conmon
+    exec: Box<dyn ExecCommand>,
+
+    #[get]
+    /// Path to the conmon binary
+    binary: PathBuf,
+}
+
+impl Conmon {
+    /// Run conmon with the provided args and return the output if the command execution succeeds.
+    /// This can still mean that conmon itself failed, which can be verified via the exist status
+    /// of the output.
+    pub async fn run(&self, args: &[Arg]) -> Result<Output> {
+        self.exec().run_output(self.binary(), args).await
+    }
+}
+
+#[derive(Clone, Default, Debug)]
+/// DefaultExecCommand is a wrapper which can be used to execute conmon in a standard way.
+struct DefaultExecCommand;
+
+impl ExecCommand for DefaultExecCommand {}
+
+#[async_trait]
+trait ExecCommand: Debug + DynClone + Send + Sync {
+    /// Run a command and return its `Output`.
+    async fn run_output(&self, binary: &Path, args: &[Arg]) -> Result<Output> {
+        Command::new(binary)
+            .args(args.iter().map(ToString::to_string))
+            .output()
+            .await
+            .context("run conmon")
+    }
+}
+
+clone_trait_object!(ExecCommand);
+
+#[derive(AsRefStr, Clone, Debug, Hash, Eq, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
```

Comment:
I'm probably missing something here. but-this-would-be-kebab-case. "RuntimeArg" and the likes look like Pascal case too me.

Reply:
Oh, that's just how we map the enum arguments to it's cli pendants, like:

`RuntimeArg` -> `--runtime-arg`

Reply:
oh. I got it! :) thank you!